source/common/upstream/upstream_impl.cc

```diff
@@ -1098,8 +1159,27 @@ ClusterInfoImpl::ClusterInfoImpl(
   }
 
   // If load_balancing_policy is set we will use it directly, ignoring lb_policy.
-  if (config.has_load_balancing_policy()) {
+  if (config.has_load_balancing_policy() ||
+      config.lb_policy() == envoy::config::cluster::v3::Cluster::LOAD_BALANCING_POLICY_CONFIG) {
     configureLbPolicies(config, server_context);
+  } else if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.convert_legacy_lb_config")) {
+    auto lb_pair = getTypedLbConfigFromLegacyProto(config);
+    load_balancer_config_ = std::move(lb_pair.first);
+    load_balancer_factory_ = lb_pair.second;
+    lb_type_ = LoadBalancerType::LoadBalancingPolicyConfig;
```

Comment:
Why we always override the lb_type to `LoadBalancerType::LoadBalancingPolicyConfig`?

Reply:
This will tell the cluster manager to create LB from the typed load balancer factory.